xarray/core/formatting.py

```diff
@@ -180,20 +199,29 @@ def format_array_flat(array, max_width):
     array that will fit within max_width characters.
     """
     # every item will take up at least two characters, but we always want to
-    # print at least one item
-    max_possibly_relevant = max(int(np.ceil(max_width / 2.0)), 1)
-    relevant_items = first_n_items(array, max_possibly_relevant)
+    # print at least first and last items
+    max_possibly_relevant = min(max(array.size, 1),
+                                max(int(np.ceil(max_width / 2.)), 2))
+    relevant_items = \
+        sum(zip_longest(
+            first_n_items(array, (max_possibly_relevant + 1) // 2),
+            reversed(last_n_items(array, max_possibly_relevant // 2))),
+            ())[:max_possibly_relevant]
```

Comment:
These lines (also below for `pprint_str`) are still pretty long. I would prefer a few more intermediate variables, and doing continuations with parentheses rather than `\`.

Also, please add a comment on the use of `sum`, since using it for concatenating tuples is unusual.